apis/v1alpha2/backendtlspolicy_types.go

```diff
@@ -0,0 +1,148 @@
+	TLS *BackendTLSPolicyConfig `json:"tls"`
+}
+
+// BackendTLSPolicyConfig contains backend TLS policy configuration.
+// +kubebuilder:validation:XValidation:message="must not contain both CertRefs and StandardCerts",rule="(has(self.certRefs) && size(self.certRefs) > 0 && has(self.standardCerts) && self.standardCerts != \"\")"
+// +kubebuilder:validation:XValidation:message="must specify either CertRefs or StandardCerts",rule="!(has(self.certRefs) && size(self.certRefs) > 0 || has(self.standardCerts) && self.standardCerts != \"\")"
+type BackendTLSPolicyConfig struct {
+	// CertRefs contains one or more references to
+	// Kubernetes objects that contain PEM-encoded TLS certificates,
+	// which are used to establish a TLS handshake between the Gateway
+	// and backend Pod.
+	//
+	// If CertRefs is empty or unspecified, then StandardCerts must
+	// be specified.  Only one of CertRefs or StandardCerts may be
+	// specified, not both.
+	//
+	// If CertRefs is empty or unspecified, then system trusted
+	// certificates should be used. If there are none, or the
+	// implementation doesn't define system trusted certificates,
+	// then a TLS connection must fail.
+	//
+	// References to a resource in a different namespace are
+	// invalid.
+	//
+	// A single CertRef to a Kubernetes ConfigMap kind has "Core"
+	// support.  Implementations MAY choose to support attaching
+	// multiple certificates to a backend, but this behavior is
+	// implementation-specific.
+	//
+	// Support: Core - An optional single reference to a Kubernetes
+	// ConfigMap, with the CA certificate in a key named `ca.crt`.
+	//
+	// Support: Implementation-specific (More than one reference, or other kinds
+	// of resources).
+	//
+	// +kubebuilder:validation:MaxItems=8
+	// +optional
+	CertRefs []v1beta1.ConfigMapObjectReference `json:"certRefs,omitempty"`
+
+	// StandardCerts specifies whether system CA certificates may
+	// be used in the TLS handshake between the gateway and
+	// backend pod.
+	//
+	// If StandardCerts is unspecified or set to "", then CertRefs must
+	// be specified with at least one entry for a valid configuration.
+	// Only one of CertRefs or StandardCerts may be specified, not both.
+	//
+	// StandardCerts must be set to "System" when CertRefs is unspecified.
+	//
+	//
+	// Support: Core for "System"
+	//
+	//
+	// +optional
+	StandardCerts *StandardCertType `json:"standardCerts,omitempty"`
+
+	// Hostname is used for two purposes in the connection between Gateways and
+	// backends:
+	//
+	// 1. Hostname MUST be used as the SNI to connect to the backend (RFC 3546).
+	// 2. Hostname MUST be used for authentication and MUST be in the certificate
+	//    served by the matching backend.
+	//
+	// Support: Core
+	Hostname v1beta1.PreciseHostname `json:"hostname"`
+}
+
+// StandardCertType is the type of CA certificate that will be used when
+// the TLS.certRefs is unspecified.
+// +kubebuilder:validation:Enum=System
```

Comment:
Yes, this is essentially a boolean with room for later expansion if we need it. I think domain-prefix is not necessary in this case.

Reply:
Yeah, I think domain-prefixed values might actually be useful in the future, but I'm fine with leaving that until later since I can't think of any immediate use cases. IMO, the most important thing at this point is that we have room for expansion here.